pkg/spoke/spokeagent.go

```diff
@@ -388,9 +413,18 @@ func (o *SpokeAgentOptions) hasValidHubClientConfig() (bool, error) {
 //
 //   1. take clusterName from input arguments if it is not empty
 //   2. TODO: read cluster name from openshift struct if the spoke agent is running in an openshift cluster
-//   3. take cluster/agent names from the mounted secret if they exist
-//   4. generate random cluster/agent names then
+//   3. take cluster/agent name from tls certification if it exist
+//   4. take cluster/agent names from the mounted secret if they exist
```

Comment:
we need a clearer doc here  that we find cluster/agent name from cert at first,  and fallback to use values in secret if it does not exists, and fall back to generate them

Reply:
Done!